builtin/logical/transit/path_encrypt.go

```diff
@@ -27,36 +44,65 @@ func (b *backend) pathEncrypt() *framework.Path {
 
 			"context": &framework.FieldSchema{
 				Type:        framework.TypeString,
-				Description: "Context for key derivation. Required for derived keys.",
+				Description: "Base64 encoded context for key derivation. Required for derived keys.",
 			},
 
 			"nonce": &framework.FieldSchema{
 				Type:        framework.TypeString,
-				Description: "Nonce for when convergent encryption is used",
+				Description: "Base64 encoded nonce for when convergent encryption is used",
 			},
 
 			"type": &framework.FieldSchema{
 				Type:    framework.TypeString,
 				Default: "aes256-gcm96",
-				Description: `When performing an upsert operation, the type of key
-to create. Currently, "aes256-gcm96" (symmetric) is the
-only type supported. Defaults to "aes256-gcm96".`,
+				Description: `
+This parameter is required when encryption key is expected to be created.
+When performing an upsert operation, the type of key to create. Currently,
+"aes256-gcm96" (symmetric) is the only type supported. Defaults to
+"aes256-gcm96".`,
 			},
 
 			"convergent_encryption": &framework.FieldSchema{
 				Type: framework.TypeBool,
-				Description: `Whether to support convergent encryption.
-This is only supported when using a key with
-key derivation enabled and will require all
-requests to carry both a context and 96-bit
-(12-byte) nonce. The given nonce will be used
-in place of a randomly generated nonce. As a
-result, when the same context and nonce are
-supplied, the same ciphertext is generated. It
-is *very important* when using this mode that
-you ensure that all nonces are unique for a
-given context. Failing to do so will severely
-impact the ciphertext's security.`,
+				Description: `
+This parameter will only be used when a key is expected to be created.  Whether
+to support convergent encryption. This is only supported when using a key with
+key derivation enabled and will require all requests to carry both a context
+and 96-bit (12-byte) nonce. The given nonce will be used in place of a randomly
+generated nonce. As a result, when the same context and nonce are supplied, the
+same ciphertext is generated. It is *very important* when using this mode that
+you ensure that all nonces are unique for a given context.  Failing to do so
+will severely impact the ciphertext's security.`,
+			},
+
+			"derived": &framework.FieldSchema{
+				Type: framework.TypeBool,
+				Description: `
+This parameter will only be used when a key is expected to be created.  Enables
+key derivation mode. This allows for per-transaction unique keys for encryption
+operations.`,
+			},
+
+			"batch": &framework.FieldSchema{
+				Type:    framework.TypeString,
+				Default: "",
+				Description: `
+Base64 encoded list of items to be encrypted in a single batch. The size of the
+input is limited to 4MB. When this parameter is set, the parameters
```

Comment:
Given that Vault now has global input size limitation on the HTTP interface, I think restricting it to 4MB here is unnecessary.